Corrfunc/mocks/DDrppi_mocks.py

```diff
@@ -219,6 +220,19 @@ def DDrppi_mocks(autocorr, cosmology, nthreads, pimax, binfile,
     weight_type : string, optional (default None)
         The type of weighting to apply.  One of ["pair_product", None].
 
+    bin_type : string, case-insensitive (default ``custom``)
+        Set to ``lin`` for speed-up in case of linearly-spaced bins.
+        In this case, the bin number for a pair separated by ``r_p`` is given by
+        ``(r_p - binfile[0])/(binfile[-1] - binfile[0])*(len(binfile) - 1)``,
+        i.e. only the first and last bins of input ``binfile`` are considered.
+        Then setting ``output_rpavg`` is virtually costless.
+        For non-linear binning, set to ``custom``.
+        ``auto`` allows for auto-detection of the binning type:
+        linear binning will be chosen if input ``binfile`` is
+        within ``rtol = 1e-05`` (relative tolerance) *and* ``atol = 1e-08``
+        (absolute tolerance) of the array
+        ``np.linspace(binfile[0], binfile[-1], len(binfile))``.
```

Comment:
The rtol/atol information is fine to include but probably not necessary.  I think I had in mind some text like: "in the vast majority of cases, `bin_type='linear'` will yield identical results to custom linear binning but with higher performance. In a few rare cases where a pair falls on a bin boundary, it may shift by one bin if using `'linear'`."

Reply:
By "custom linear binning", do you mean bin_type = 'custom' and linear binning provided in binfile?
In the case the user wants to use truly linear binning, bin_type='linear' is actually probably the more correct way to go? (i.e. we do not want the user to think bin_type='linear' is an approximation in this case)

Reply:
Yes, that's what I meant. And I agree we don't want users to think this is an approximation!  I was thinking the custom binning is exact, in that we use an inequality check of the bin edges with the separation, whereas the multiplication by `1/bin_size` to get the bin index might be inexact, because it is not guaranteed that `1/bin_size` is exactly representable in IEEE 754.  But now that I think about it more, it's not guaranteed that the bin edges are exactly representable either.

So we still are left in the case where `bin_type='linear'` and `bin_type='custom'` may yield different results, and that's okay.  Neither is more correct, and the differences will always be small, but perhaps it's prudent to inform users of this. Something like: "In the vast majority of cases, bin_type='linear' will yield identical results to custom linear binning but with higher performance. In a few rare cases where a pair falls on a bin boundary, 'linear' and custom linear may disagree on which bin the pair falls into due to finite floating point precision."

Reply:
Thanks for the feedback, that totally makes sense and I agree with the reformulation.
Do you want me to remove the note about rtol and atol as criteria to switch for linear binning in the bin_type='auto' case?
(I added it was this is specified somewhere in the C code where a Python user may never go to, but can instead reference to utils/utils.c/detect_bin_type for the interested user to look at)

Reply:
I would probably take it out myself just to prevent the docstring from getting overly long, but I also don't mind if you want to leave it in.  In any case, I think since the default is `custom` (at least for now), very few people will use `auto`; they'll just switch straight to `linear` if they have need for it.

On that note, I wouldn't mind if we switched the default to `auto`, but I can see pros and cons to either `auto` or `custom` as the default.